File: Codes in C++/Graph/Distance to nearest 1 in binary matrix.cpp
vector < vector < int > > nearest(vector < vector < int >> & mat, int n, int m) {
    vector < vector < int > > vis(n,vector<int>(m,0));
    vector < vector < int > > dis(n,vector<int>(m,0));
    queue<pair<pair<int,int>,int>> q;

    for(int i=0;i<n;i++)
    {
        for(int j=0;j<m;j++)
        {
            if(mat[i][j]==1)
            {
                q.push({{i,j},0});
                vis[i][j]=1;
            }
            else
                vis[i][j]=0;
        
        }
    }
    while(!q.empty())
    {
        int row=q.front().first.first;
        int col=q.front().first.second;
        int step=q.front().second;
        q.pop();
        int dr[]={-1,0,+1,0};
        int dc[]={0,+1,0,-1};
        dis[row][col]=step;
        for(int i=0;i<4;i++)
        {
            int nrow=row+dr[i];
            int ncol=col+dc[i];
            if(nrow>=0 && nrow<n && ncol>=0 && ncol<m && vis[nrow][ncol]==0 )
            {
                vis[nrow][ncol]=1;
                q.push({{nrow,ncol},step+1});
            }
        }
    }

    return dis;
}

File: Codes in Python3/Graph/Distance to nearest 1 in binary matrix.py
def nearest(mat,n,m):
    vis=[[0]*m for _ in range(n)]
    dis=[[0]*m for _ in range(n)]
    q=[]
    for i in range(n):
        for j in range(m):
            if mat[i][j]==1:
                q.append((i,j,0))
                vis[i][j]=1
            
    while q:
        row,col,step=q.pop(0)
        dr=[-1,0,+1,0]
        dc=[0,+1,0,-1]
        dis[row][col]=step
        for i in range(4):
            nrow=row+dr[i]
            ncol=col+dc[i]
            if nrow>=0 and nrow<n and ncol>=0 and ncol<m and vis[nrow][ncol]==0:
                vis[nrow][ncol]=1
                q.append((nrow,ncol,step+1))
    return dis
    pass
